### PR TITLE
Migrated to Spring Boot 3

### PR DESCRIPTION
**Pull request by mstrankowski**
compiles, I was forced to update to httpclient5 from apache. Need to verify no impact on ssl mechanism after changes

### CODE REVIEW

File: deprecated/alfresco-transformer-base/src/main/java/org/alfresco/transformer/config/MTLSConfig.java [hunk 1]
```diff
@@ -120,11 +125,23 @@ private boolean isKeystoreConfigured()
     }
 
     private RestTemplate createRestTemplateWithSslContext(SSLContextBuilder sslContextBuilder) throws NoSuchAlgorithmException, KeyManagementException {
-        SSLContext sslContext = sslContextBuilder.build();
-        SSLConnectionSocketFactory sslContextFactory = hostNameVerificationDisabled ? new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE)
-                : new SSLConnectionSocketFactory(sslContext);
+        final SSLConnectionSocketFactoryBuilder sslConnectionSocketFactoryBuilder =
+                SSLConnectionSocketFactoryBuilder.create()
+                        .setSslContext(sslContextBuilder.build())
+                        .setTlsVersions(TLS.V_1_2, TLS.V_1_3);
```
Comment (mstrankowski) [inline deprecated/alfresco-transformer-base/src/main/java/org/alfresco/transformer/config/MTLSConfig.java:131]: We have such restriction implemented in Repo, so adding it here would make it consistent in my opinion.

File: deprecated/alfresco-transformer-base/src/main/java/org/alfresco/transformer/config/MTLSConfig.java [hunk 2]
```diff
@@ -120,11 +125,23 @@ private boolean isKeystoreConfigured()
...
+            sslConnectionSocketFactoryBuilder.setHostnameVerifier(NoopHostnameVerifier.INSTANCE);
+        }
+        final SSLConnectionSocketFactory sslConnectionSocketFactory = sslConnectionSocketFactoryBuilder.build();
+
+        final Registry<ConnectionSocketFactory> sslSocketFactoryRegistry =
+                RegistryBuilder.<ConnectionSocketFactory> create()
+                        .register("https", sslConnectionSocketFactory)
+                        .build();
+
+        final BasicHttpClientConnectionManager sslConnectionManager = new BasicHttpClientConnectionManager(sslSocketFactoryRegistry);
```
Comment (mstrankowski) [inline deprecated/alfresco-transformer-base/src/main/java/org/alfresco/transformer/config/MTLSConfig.java:142]: It's either this - basic, or poolable. I don't think we use poolable "perks" in transform-core at all, so I chose a basic option.

File: engines/base/pom.xml [hunk 3]
```diff
@@ -88,13 +88,8 @@
...
         <dependency>
-            <groupId>org.apache.httpcomponents</groupId>
-            <artifactId>httpclient</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.httpcomponents</groupId>
-            <artifactId>httpmime</artifactId>
-            <scope>test</scope>
+            <groupId>org.apache.httpcomponents.client5</groupId>
+            <artifactId>httpclient5</artifactId>
```
Comment (mstrankowski) [inline engines/base/pom.xml:92]: Forced to update by spring 6 backing spring boot 3

File: engines/base/src/main/java/org/alfresco/transform/base/config/MTLSConfig.java [hunk 4]
```diff
@@ -179,11 +184,23 @@ private HttpClient createHttpClientWithSslContext(SslContextBuilder sslContextBu
     }
 
     private RestTemplate createRestTemplateWithSslContext(SSLContextBuilder sslContextBuilder) throws NoSuchAlgorithmException, KeyManagementException {
-        SSLContext sslContext = sslContextBuilder.build();
-        SSLConnectionSocketFactory sslContextFactory = hostNameVerificationDisabled ? new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE)
-                : new SSLConnectionSocketFactory(sslContext);
+        final SSLConnectionSocketFactoryBuilder sslConnectionSocketFactoryBuilder =
```
Comment (mstrankowski) [inline engines/base/src/main/java/org/alfresco/transform/base/config/MTLSConfig.java:187]: Exact same change as above in similar context. We just have a deprecated copy of code and a new one.

File: engines/base/src/main/resources/templates/test.html [hunk 5]
```diff
@@ -34,15 +34,15 @@ <h2 th:text="${title}"></h2>
                         <option value="text/plain">txt</option>
                     </select></td></tr>
 
-                <th:block th:each="i: ${#numbers.sequence(0, T(java.lang.Math).min(18, transformOptions.size) - 1)}">
+                <th:block th:each="i: ${#numbers.sequence(0, T(java.lang.Math).min(18, transformOptions.size()) - 1)}">
```
Comment (mstrankowski) [inline engines/base/src/main/resources/templates/test.html:37]: SpEL had issues without brackets, when it tries to reach through parameter name it seems to go through reflection that is limited by context settings, with brackets it just calls the method without issues.

File: engines/base/src/test/java/org/alfresco/transform/base/clients/SfsClient.java [hunk 6]
```diff
@@ -71,17 +72,17 @@ public static String uploadFile(final String fileToUploadName, final String sfsB
             sfsBaseUrl+"/alfresco/api/-default-/private/sfs/versions/1/file");
         post.setEntity(MultipartEntityBuilder
             .create()
-            .setMode(HttpMultipartMode.BROWSER_COMPATIBLE)
+            .setMode(HttpMultipartMode.LEGACY)
```
Comment (mstrankowski) [inline engines/base/src/test/java/org/alfresco/transform/base/clients/SfsClient.java:75]: Based on official documentation description for both of these options, they should be the same.

File: pom.xml [hunk 7]
```diff
@@ -220,14 +220,9 @@
...
-                <artifactId>httpclient</artifactId>
-                <version>4.5.14</version>
-            </dependency>
-            <dependency>
-                <groupId>org.apache.httpcomponents</groupId>
-                <artifactId>httpmime</artifactId>
-                <version>4.5.14</version>
+                <groupId>org.apache.httpcomponents.client5</groupId>
+                <artifactId>httpclient5</artifactId>
+                <version>5.2</version>
```
Comment (mstrankowski) [inline pom.xml:225]: httpmime seems to be contained in httpclient5

**Review by kmagdziarz — APPROVED** [review]

**Pull request merged**